agent.go

```diff
@@ -173,8 +186,18 @@ func (agent *agentS) SendSpans(spans []Span) error {
 
 	_, err := agent.request(agent.makeURL(agentTracesURL), "POST", spans)
 	if err != nil {
-		agent.logger.Error("failed to send spans to the host agent: ", err)
-		agent.reset()
+		if err == payloadTooLargeErr {
+			agent.printPayloadTooLargeErrInfoOnce.Do(
+				func() {
+					agent.logDetailedInformationAboutDroppedSpans(5, spans, err)
```

Comment:
Query: What is the significance of 5 here as first argument?

Reply:
Good question: this is a number of spans we want to log in this case. You are right is a kind of "magic constant". Maybe it is a good idea to extract it and give it a name.